staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -362,11 +471,11 @@ func NewEventCorrelator(clock clock.Clock) *EventCorrelator {
 
 // EventCorrelate filters, aggregates, counts, and de-duplicates all incoming events
 func (c *EventCorrelator) EventCorrelate(newEvent *v1.Event) (*EventCorrelateResult, error) {
+	aggregateEvent, ckey := c.aggregator.EventAggregate(newEvent)
```

Comment:
event aggregate is cheap.  we had it lower before because i had not thought through spam detection well enough when it was first written.  we need spam filtering after the event we plan to send to the server is aggregated so spam filtering works on aggregated event itself.